### DISCUSSION TITLE
通过DDL创建表

### DISCUSSION THREAD

**Discussion by imotai** [4paradigm/OpenMLDB#195] [Ideas]
# OpenMLDB支持类型

|#|类型名称|说明|
|:----|:----|:----|
|1|bigint|64位int|
|2|int|32位int|
|3|smallint|16位int|
|4|tinyint|8位int|
|5|boolean|bool类型|
|6|date|0001-01-01 to 9999-12-31|
|7|double|双精度浮点数|
|8|float|单精度浮点数|
|11|decimal|    |
|12|string|字符串|
|13|timestamp|时间戳|

# DDL 例子

```sql
CREATE TABLE testdb.openmldb_iceberg (
    id BIGINT,
    time TIMESTAMP,
    s STRING,
    b BOOLEAN,
    index(ts=time)
);
```
参考[https://impala.apache.org/docs/build/html/topics/impala_create_table.html#create_table](https://impala.apache.org/docs/build/html/topics/impala_create_table.html#create_table)
1. 兼容if not exist 语法

## 索引介绍

ddl中必须指定时间索引，openmldb要求数据表必须包含时间索引，时间索引使用的列可以是
* 时间戳
* 日期

# 命令行交互方式
### 非交互方式

```shell
openmldb --sql test.ddl --uri xxxxx
```
### 交互方式

不进入database可以

```shell
openmldb --uri xxxxxxx
>CREATE TABLE testdb.openmldb_iceberg (
+id BIGINT,
+time TIMESTAMP,
+s STRING,
+b BOOLEAN,
index(ts=time));
```

进入database可以是

```shell
openmldb --uri xxxxxxx
>use testdb;
>CREATE TABLE openmldb_iceberg (
+id BIGINT,
+time TIMESTAMP,
+s STRING,
+b BOOLEAN,
+index(ts=time)) ;
```

**Comment by tobegit3hub** [comment]
`index`语法主要是为了定义分区键，可以考虑直接用相对标准的`partitioned by`语法。

```
CREATE TABLE testdb.openmldb_iceberg (
    id BIGINT,
    time TIMESTAMP,
    s STRING,
    b BOOLEAN
) PARTITIONED BY (
  time TIMESTAMP
)
```
